src/net/tcp/stream.rs

```diff
@@ -149,6 +149,34 @@ impl TcpStream {
         (Ok(()), buf)
     }
 
+    /// Write data from buffers into this socket returning how many bytes were
+    /// written.
+    ///
+    /// This function will attempt to write the entire contents of `bufs`, but
+    /// the entire write may not succeed, or the write may also generate an
+    /// error. The bytes will be written starting at the specified offset.
+    ///
+    /// # Return
+    ///
+    /// The method returns the operation result and the same array of buffers
+    /// passed in as an argument. A return value of `0` typically means that the
+    /// underlying socket is no longer able to accept bytes and will likely not
+    /// be able to in the future as well, or that the buffer provided is empty.
+    ///
+    /// # Errors
+    ///
+    /// Each call to `write` may generate an I/O error indicating that the
+    /// operation could not be completed. If an error is returned then no bytes
+    /// in the buffer were written to this writer.
```

Comment:
This comment about no bytes in the buffer seems suspect. The writev is of a list of buffers, the comment makes it sound like a single buffer. I don't see a kernel man page for writev making this kind of statement. But this comment was already made for the File writev_at function so no need to change it here. Just sayin, it looks odd.